### PR TITLE
#169257412 Builds Avail requests for approval

### PR DESCRIPTION
**Pull request by b0nbon1**
#### What does this PR do?
It filters the requests and fetches the open requests from users to their line manager for approval.
#### Description of Task to be completed?
- [X] fetch requests for approval to the direct report manager
- [X] check the line manager
- [X] filter the requests with status open
#### How should this be manually tested?
- git clone the repo
- run `npm install` to install dependencies
- run `npm run migrate` to update the tables
- run signup two users with the data on the `POST {{url}}/api/v1/auth/signup`
- update one user to be a manager by using `Patch {{url}}/api/v1/auth/user/role`
- update the user using the line manager id `Patch {{url}}/api/v1/user/update-profile`
- using the user token create a trip
- using line manager token get the trips using endpoint `Get {{url}}/api/v1/requests/manager`
### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#169257412](https://www.pivotaltracker.com/n/projects/2407417/stories/169257412)
#### Screenshots 
image

### CODE REVIEW

File: src/middlewares/roleAuthorization.js [hunk 1]
```diff
@@ -6,8 +6,16 @@ const roleAuthorization = {
     if (role === 'super_administrator') {
       next();
     } else {
-      Responses.handleError(403, 'insufficient privileges', res);
+      return Responses.handleError(403, 'insufficient privileges', res);
     }
+  },
+
+  checkIsManager(req, res, next) {
```
Comment (gildniy) [inline src/middlewares/roleAuthorization.js:13]: I think these 2 check methods can be done into one and pass the allowed roles name array in params

**Review by gildniy — CHANGES_REQUESTED** [review]

File: src/controllers/request.controller.js [hunk 2]
```diff
@@ -34,6 +34,18 @@ class Requests {
     }
     return Responses.handleSuccess(200, 'Succesfully fetched the request', res, requests);
```
Comment (gildniy) [inline src/controllers/request.controller.js]: I think `Succesfully` might be `Successfully`!

Comment (gildniy) [inline src/controllers/request.controller.js]: I think `Succesfully` might be `Successfully`!

File: src/controllers/request.controller.js [hunk 3]
```diff
@@ -34,6 +34,18 @@ class Requests {
...
+  /**
+   *
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} succesful fetched the requests
+   */
+  async getLineManagerRequest(req, res) {
+    const currentUser = res.locals.user;
+    const requests = await getManagerRequest(currentUser.userId);
+    return Responses.handleSuccess(200, 'Succesfully fetched the requests', res, requests);
```
Comment (gildniy) [inline src/controllers/request.controller.js]: I think `Succesfully` might be `Successfully`!

**Review by gildniy — CHANGES_REQUESTED** [review]

**Review by gildniy — APPROVED** [review]
Looks great now, I think it can be merged.

**Review by Mastel22 — APPROVED** [review]
Great job.

**Comment by nshaphan** [comment]
Great work, but you would edit Pull request description to use actual endpoint instead of links to the postman workspace.
I recommend you also to use the team workspace. To have all endpoint at one place

**Review by nshaphan — APPROVED** [review]

**Pull request merged**